src/discord/discord.service.ts

```diff
@@ -21,6 +34,24 @@ export class DiscordService implements OnModuleInit {
         await interaction.reply('Pong!');
       }
     });
+
+    this.client.on('messageCreate', (msg) => {
+      if (msg.author.bot) return;
+
+      const isFromTestChannel =
+        msg.channelId ===
+        this.configService.getOrThrow('DISCORD_BOT_TESTING_CHANNEL_ID');
+      const isProduction = this.configService.get('NODE_ENV') === 'production';
+
+      if (msg.channel.type === 0 && (isFromTestChannel || isProduction)) {
+        this.textChannelHandler(msg);
+      }
+
+      if (msg.channel.type === 1) {
+        this.dmHandler(msg);
+      }
+    });
```

Comment:
I have started implementing this change. I recently read an article that mixing catch and async/await can cause issues in some cases. I'm not sure if this could be the case here, but maybe you can have a look? https://maximorlov.com/why-you-shouldnt-mix-promise-then-with-async-await/

Reply:
The article is right in general IMO but in this case we don't have much choice and we have isolated the usage of promise chaining to a single method.